google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/project/ProjectSelector.java

```diff
@@ -73,6 +87,22 @@ public void setSelectedProject(CloudProject cloudProject) {
     }
   }
 
+  /**
+   * Loads active {@link CloudProject}, if IDE {@link Project} has been specified. This should be
+   * called when project selector and the related UI are completely initialized. Calls listeners to
+   * notify on new project selection.
+   */
+  public void loadActiveCloudProject() {
```

Comment:
so the calls to `loadActiveCloudProject` calls get invoked separately from the calls to `setSelectedProject` which typically get called from platform managed methods like `resetEditorFrom`. Is there a possibility for UI flickering here (first showing the active project, then updating it to the panel specific persisted project)?

Reply:
also, is there any case where we wouldn't want to load the active/default project?

I'm wondering if we can make the call to `loadActiveCloudProject` implicit in the ProjectSelector itself rather than having the clients call it. In other words, have the ProjectSelector do this itself as part of initialization.

Reply:
Depends on particular UI design and how long it takes to update it. We can delay `loadActiveCloudProject` and do the explicit check in `resetEditorFrom` for such forms (run/deployment and cloud debugger) to make sure we don't do double work.

Reply:
my main interest with this is really to see if we can have the ProjectSelector handle all default/active project logic itself rather than having the clients have to explicitly call it (whether it be on init or in `resetEditorFrom`). 

for instances of the selector where we wouldn't want to show an active project (if such a case exists), I guess we could flag this on initialization of the project selector.

Reply:
As discussed offline, it could be complicated to properly define a moment when default project needs to be loaded without knowing details of surrounding UI initialization. I'll change forms with their own selected project not to load default/active project without checking on existing project first.